src/popup/components/Protocol/DataScreen/Minting.tsx

```diff
@@ -42,22 +49,103 @@ export function Minting() {
       </div>,
     );
 
+  const [registering, setRegistering] = useState(false);
+  const tryRegister = async () => {
+    try {
+      setRegistering(true);
+      await getMintingRegistrar().tryRegister();
+    } finally {
+      isRegistered.reload();
+      setRegistering(false);
+    }
+  };
+
   return (
-    <Hero
-      title="Minting"
-      callout={<Activated text={callout[0]} icon={callout[1]} />}
-    >
-      <HistoryContainer>{historyItems}</HistoryContainer>
+    <Hero title="Minting" callout={callout}>
+      <HistoryContainer>
+        {showRegisterButton ? (
+          <Button onClick={tryRegister} loading={registering}>
+            Register
+          </Button>
+        ) : null}
+        {historyItems}
+      </HistoryContainer>
     </Hero>
   );
 }
 
+const NotRegisteredContainer = styled.div`
+  .text p {
+    text-decoration: underline;
+    text-decoration-style: dotted;
+  }
+
+  #notRegistered {
+    opacity: 1;
+  }
+`;
+
+function NotRegistered() {
+  const error = useLoadedState(async () => {
+    return await getMintingRegistrar().latestError();
+  });
+
+  const errorMessage = error
+    .map((e) => <MintingError error={e} />)
+    .unwrapOrDefault(<p>Loading</p>);
+
+  return (
+    <NotRegisteredContainer>
+      <div className="text" data-tip data-for="notRegistered">
+        <Activated text="Not Registered" icon={IconNames.INVALID} />
+      </div>
+
+      <ReactTooltip id="notRegistered" place="top" effect="solid">
```

Comment:
We can maybe get the effect of top left with offset. Check this thread - https://stackoverflow.com/questions/59826015/how-to-change-position-of-reacttooltip

Reply:
Possibly, but I don't really trust the position of the library. Even with "top" it is showing up to the left. So I'm okay with letting it be wherever it's going to be. Adding an offset just increases the chance it shows up somewhere stupid.